dnf5daemon-server/session.cpp

```diff
@@ -163,20 +157,25 @@ bool Session::read_all_repos() {
         return false;
     }
     repositories_status = dnfdaemon::RepoStatus::PENDING;
-    //auto & logger = base->get_logger();
-    libdnf::repo::RepoQuery enabled_repos(*base);
-    enabled_repos.filter_enabled(true);
-    enabled_repos.filter_type(libdnf::repo::Repo::Type::AVAILABLE);
-    for (auto & repo : enabled_repos) {
-        repo->set_callbacks(std::make_unique<DbusRepoCB>(*this));
-    }
 
     bool retval = true;
 
-    try {
-        base->get_repo_sack()->update_and_load_enabled_repos(false);
-    } catch (const std::runtime_error & ex) {
-        retval = false;
+    if (session_configuration_value<bool>("load_available_repo", true)) {
+        //auto & logger = base->get_logger();
+        libdnf::repo::RepoQuery enabled_repos(*base);
+        enabled_repos.filter_enabled(true);
+        enabled_repos.filter_type(libdnf::repo::Repo::Type::AVAILABLE);
+        for (auto & repo : enabled_repos) {
+            repo->set_callbacks(std::make_unique<DbusRepoCB>(*this));
+        }
+
+        try {
+            base->get_repo_sack()->update_and_load_enabled_repos(session_configuration_value<bool>("load_system_repo", true));
+        } catch (const std::runtime_error & ex) {
+            retval = false;
+        }
+    } else if (session_configuration_value<bool>("load_system_repo", true)) {
+        base->get_repo_sack()->get_system_repo()->load();
```

Comment:
Let's discuss the API around loading repositories. To allow multiple points of repository load will create problem with application of modularity rules. We must prevent user to load multiple times any repository including system repository.